perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_eduPersonScopedAffiliations.java

```diff
@@ -20,4 +45,73 @@ public String getDestinationAttributeFriendlyName() {
 		return "eduPersonScopedAffiliations";
 	}
 
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition destinationAttributeDefinition) throws InternalErrorException {
+		//get already filled value obtained from UserExtSources
+		Attribute attribute = super.getAttributeValue(sess, user, destinationAttributeDefinition);
+
+		Attribute destinationAttribute = new Attribute(destinationAttributeDefinition);
+		//get values previously obtained and add them to Set representing final value
+		//for values use set because of avoiding duplicities
+		Set<String> valuesWithoutDuplicities = new HashSet<>();
+		if (attribute != null) {
+			Object value = attribute.getValue();
+			if (value != null && value instanceof Set) {
+				Object[] valArray = ((Set) value).toArray();
+				if (valArray instanceof String[]) {
+					for (String val: (String[]) valArray) {
+						valuesWithoutDuplicities.add(val);
+					}
+				}
+			}
+		}
```

Comment:
This is really complicated code. I would prefer something more human-readable. You can work with situation, that super implementation of method getAttributeValue never return null instead of Attribute and also the value (if it is not-null) is always instance of ArrayList (not Set).

Reply:
The Attribute class has the method valueAsList() which returns its value as ArrayList<String>.